comp/dogstatsd/server/server.go

```diff
@@ -523,6 +542,11 @@ func (s *server) handleMessages() {
 		go worker.run()
 		s.workers = append(s.workers, worker)
 	}
+
+	// create the metric names blocklist, one per worker
+	// -----------
+
+	s.SetBlocklist(s.config.GetStringSlice("statsd_metric_blocklist"), s.config.GetBool("statsd_metric_blocklist_match_prefix"))
```

Comment:
So we set the initial blocklist to the value of the existing blocklist configuration parameters.. and then any RC-based updates would be sent to update the worker, which makes sense.

However, the main question I'm left with: is it intended that the RC-based blocklist _replaces_ the manually configured one?

Reply:
Fair, I have to think about this one.

Reply:
Shipped in https://github.com/DataDog/datadog-agent/pull/36761/commits/9d4a81ceb42616d4e2b761eadeebef0326587547